src/labone/mock/automatic_hpk_functionality.py

```diff
@@ -0,0 +1,148 @@
+import asyncio
+import fnmatch
+import re
+import time
+import typing as t
+from typing import Any, Coroutine
+
+from labone.core import ListNodesFlags, ListNodesInfoFlags
+from labone.core.helper import LabOneNodePath
+from labone.core.session import NodeInfo
+from labone.core.subscription import StreamingHandle
+from labone.core.value import AnnotatedValue
+from labone.mock.hpk_functionality import HpkMockFunctionality
+
+
+class AutomaticHpkFunctionality(HpkMockFunctionality):
+    """Already predifined behaviour:
+    - simulating state for get/set
+    - adding chronological timestamps to responses
+    - manages subscriptions and passes all changes into the queues.
+    - answering list_nodes(_info) via knowledge of the tree structure
+    - reducing get_with_expression/set_with_expression naturally to get/set
+      (via knowledge of the tree structure)
+    """
+
+    def __init__(self, paths_to_info: dict[LabOneNodePath, NodeInfo] = {}) -> None:
+        self._paths_to_info = paths_to_info
+        self._memory = {}
+        self._path_to_streaming_handles: dict[
+            LabOneNodePath,
+            list[StreamingHandle],
+        ] = {}
+
+    def get_timestamp(self) -> int:
+        return time.clock_gettime_ns(time.CLOCK_MONOTONIC)
+
+    async def list_nodes_info(
+        self,
+        path: LabOneNodePath = "",
+        *,
+        flags: ListNodesInfoFlags | int = ListNodesInfoFlags.ALL,
+    ) -> Coroutine[Any, Any, dict[LabOneNodePath, NodeInfo]]:
+        """
+        Warning:
+            Flags will be ignored in this implementation.
```

Comment:
I agree that for a first implementation this is out of scope ... but I would make this a TODO ... We should support it eventually. we have most the information we need from the path_to_info dict.